### PR TITLE
chore: add semgrep scanning to PR & manual workflow

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

add semgrep code scanning workflow

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/pr_check.yml [hunk 1]
```diff
@@ -11,6 +11,11 @@ env:
   TEST_SSL_KEY: ${{ secrets.TEST_SSL_KEY }}
 
 jobs:
+  semgrep:
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/pr_check.yml]: To make the new job more consistency with others, you should update the following list below.
1. rename job property `semgrep` to scan-semgrep`
2. rename job `scan` to `scan-sonarcloud`
3. add property `name: Scan / Semgrep`
4. move `scan-semgrep` job after the `scan-sonarcloud` job.

Comment (wattachai-lseg) [reply .github/workflows/pr_check.yml]: updated to your suggestion.

File: .github/workflows/run_semgrep_scan.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        description: verbose flag for semgrep scan
+        type: boolean
+        default: false
+
+jobs:
+  semgrep:
+    uses: ./.github/workflows/semgrep.yml
+    with:
+      verbose: ${{ inputs.verbose }}
+    secrets: inherit
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/run_semgrep_scan.yml:18]: Should pass only one secret `SEMGREP_APP_TOKEN` not inherit all.

File: .github/workflows/semgrep.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      # A Docker image with Semgrep installed. Do not change this.
+      image: returntocorp/semgrep
+
+    steps:
+      # Fetch project source with GitHub Actions Checkout.
+      - uses: actions/checkout@v3
+      # Run the "semgrep ci" command on the command line of the docker image.
+      - name: setup scan option
+        run: |
+          [[ "${{ inputs.verbose }}" == "true" ]] && export VERBOSE=--verbose || export FORCE=''
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/semgrep.yml]: Why need to set the`FORCE=''` env?

File: .github/workflows/semgrep.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  workflow_call:
+    inputs:
+      verbose:
+        description: verbose flag for semgrep scan
+        type: boolean
+        default: false
+
+jobs:
+  semgrep:
+    name: semgrep/ci
```
Comment (Sakchai-Refinitiv) [inline .github/workflows/semgrep.yml]: Rename to `name: Scan / Semgrep`

File: .github/workflows/run_semgrep_scan.yml (hunk 2 above)
Comment (wattachai-lseg) [reply .github/workflows/run_semgrep_scan.yml:18]: we use `secrets: inherit` in many locations such as prod_release. What's bad about this inherit?

File: .github/workflows/semgrep.yml (hunk 3 above)
Comment (wattachai-lseg) [reply .github/workflows/semgrep.yml]: oops, my bad. fixed.

File: .github/workflows/semgrep.yml (hunk 4 above)
Comment (wattachai-lseg) [reply .github/workflows/semgrep.yml]: updated

File: .github/workflows/run_semgrep_scan.yml (hunk 2 above)
Comment (Sakchai-Refinitiv) [reply .github/workflows/run_semgrep_scan.yml:18]: It is about security issues by sending all secret keys or tokens to third-party services.

Comment (Sakchai-Refinitiv) [reply .github/workflows/run_semgrep_scan.yml:18]: I read the document GitHub limit accessing secret in the organization or enterprise scope, so we can use inherit.

**Review by Sakchai-Refinitiv — APPROVED** [review]
Good feature

**Pull request merged**